dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMain.java

```diff
@@ -145,4 +77,112 @@ public static void main(String[] arguments)
       System.err.println(ex.getMessage());
     }
   }
+
+  private static ImmutableSet<LinkageProblem> checkJarFiles(
+      LinkageCheckerArguments linkageCheckerArguments)
+      throws IOException, TransformerException, XMLStreamException {
+    ImmutableList<ClassPathEntry> inputClassPath = linkageCheckerArguments.getJarFiles();
+    ImmutableSet<ClassPathEntry> entryPoints = ImmutableSet.copyOf(inputClassPath);
+    LinkageChecker linkageChecker =
+        LinkageChecker.create(
+            inputClassPath, entryPoints, linkageCheckerArguments.getInputExclusionFile());
+    ImmutableSet<LinkageProblem> linkageProblems = linkageChecker.findLinkageProblems();
+
+    linkageProblems = filterReachable(linkageCheckerArguments, linkageChecker, linkageProblems);
+
+    if (writeExclusionFile(linkageCheckerArguments, linkageProblems)) {
+      return ImmutableSet.of();
+    }
+
+    if (!linkageProblems.isEmpty()) {
+      System.out.println(LinkageProblem.formatLinkageProblems(linkageProblems));
+    }
+
+    return linkageProblems;
+  }
+
+  private static ImmutableSet<LinkageProblem> filterReachable(
+      LinkageCheckerArguments linkageCheckerArguments,
+      LinkageChecker linkageChecker,
+      ImmutableSet<LinkageProblem> linkageProblems) {
+    if (linkageCheckerArguments.getReportOnlyReachable()) {
+      ClassReferenceGraph graph = linkageChecker.getClassReferenceGraph();
+      linkageProblems =
+          linkageProblems.stream()
+              .filter(
+                  (LinkageProblem problem) ->
+                      graph.isReachable(problem.getSourceClass().getBinaryName()))
+              .collect(toImmutableSet());
+    }
+    return linkageProblems;
+  }
+
+  /**
+   * Writes {@code linkageProblems} to a file and returns true if {@code linkageCheckerArguments}
+   * specifies the option; otherwise returns false.
+   */
+  private static boolean writeExclusionFile(
```

Comment:
@elharo Do you think of a better name for such function that does nothing depending on the argument? I was thinking `writeExclusionFileIfnecessary` but this sounds redundant.

Reply:
writeExclusionFile is fine